### ISSUE TITLE
YUM CACHE Failure for 'influxdb install'

### ISSUE THREAD

**Issue by EZtheOG** [bdangit/chef-influxdb#114]
I'm on AWS using their centos brand - which maybe is the issue? - but when I run a kitchen converge in AWS EC2 I get that the influxdb yum repo is unavailable.

 `(     STDERR: https://repos.influxdata.com/centos/latest/x86_64/stable/repodata/repomd.xml: [Errno 14] PYCURL ERROR 22 - "The requested URL returned error: 404 Not Found"
       Trying other mirror.`
       `STDERR: yum-dump Repository Error: failure: repodata/repomd.xml from influxdb: [Errno 256] No more mirrors to try.
       ---- End output of /usr/bin/python2.7 /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/provider/package/yum/yum-dump.py --options --installed-provides --yum-lock-timeout 30 ----
       Ran /usr/bin/python2.7 /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/provider/package/yum/yum-dump.py --options --installed-provides --yum-lock-timeout 30 returned 1)`

Even if I install the REPO using the yum cookbook - still nothing.

----- 
TL;DR something is wrong with the repo (?) and I wasn't sure who to contact.

**Comment by EZtheOG** [comment]
**Curling proves that the `latest` part of the URL seems to be the problem:**

`$ curl https://repos.influxdata.com/centos/latest/x86_64/stable/repodata/repomd.xml`
`<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">`
`<html><head>`
`<title>404 Not Found</title>`
`</head><body>`
`<h1>Not Found</h1>`
`<p>The requested URL /centos/latest/x86_64/stable/repodata/repomd.xml was not found on this server.</p>`
`<hr>
<address>Apache/2.4.7 (Ubuntu) Server at repos.influxdata.com Port 443</address>
</body></html>`

**where as if you point to a specific distro (e.g. 6Server):**
`$ curl https://repos.influxdata.com/centos/6Server/x86_64/stable/repodata/repomd.xml
<?xml version="1.0" encoding="UTF-8"?>
<repomd xmlns="http://linux.duke.edu/metadata/repo" xmlns:rpm="http://linux.duke.edu/metadata/rpm">
 <revision>1470648776</revision>
<data type="filelists">
.......
`

**Comment by bdangit** [comment]
hrmm looks like chef decided that `node['platform_version']` on Amazon AMIs would be "latest". 

ref: https://github.com/bdangit/chef-influxdb/blob/master/attributes/default.rb#L125

It would help if you could you print out ohai and grep for platform_version? and anything related to platform?

Also what version of chef-client and ohai are you running?

**Comment by EZtheOG** [comment]
Hmmmm interesting.
I am running Chef Client: 12.12.15
and Ohai: 8.17.1

Here is the requested OHAI info.
```ohai | grep platform*
  "platform": "amazon",
  "platform_version": "2016.03",
  "platform_family": "rhel",
      "platform": "x86_64-linux",```

**Comment by bdangit** [comment]
thanks @EZtheOG.  Might have to try a different way to get the correct platform_version.

Can you try running these commands:

`lsb_release -rs | cut -f1 -d.`

`cat /etc/redhat-release`

**Comment by EZtheOG** [comment]
no problem @bdangit. So, ```/etc/redhat-release``` doesn't exist on this AMZN AMI. But, here is what I have:
```[ec2-user@influxdb ~]$ lsb_release -a```
```LSB Version:	:base-4.0-amd64:base-4.0-noarch:core-4.0-amd64:core-4.0-noarch```
```Distributor ID:	AmazonAMI```
```Description:	Amazon Linux AMI release 2016.03```
```Release:	2016.03```
```Codename:	n/a```

and : 
```[ec2-user@influxdb ~]$ uname -a```
```Linux platform-influxdb-001.p.use1.risingtidegames.com 4.4.11-23.53.amzn1.x86_64 #1 SMP Wed Jun 1 22:22:50 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux```

**Comment by EZtheOG** [comment]
Also, to be clear, I am using Amazon's Centos V and not Amazon's RedHat Server:
```Amazon Linux AMI 2016.03.3 (HVM), SSD Volume Type - ami-31490d51```

**Comment by bdangit** [comment]
Interesting. So question to you, how do you know if you have Centos 6 or Centos 7?

**Comment by bdangit** [comment]
oh btw, since you're running centos, you should have `/etc/centos-release`

**Comment by EZtheOG** [comment]
Well, the only reason why I know it is NOT Centos7 is because Centos 7 uses systemd -- which I don't have.

Also, Amazon doesn't use ```/etc/centos-release``` as it's place for info (after reading some AWS documents) it is in ```/etc/os-release```:

```[ec2-user@influxdb ~]$ cat /etc/os-release```
```NAME="Amazon Linux AMI"```
```VERSION="2016.03"```
```ID="amzn"```
```ID_LIKE="rhel fedora"```
```VERSION_ID="2016.03"```
```PRETTY_NAME="Amazon Linux AMI 2016.03"```
```ANSI_COLOR="0;33"```
```CPE_NAME="cpe:/o:amazon:linux:2016.03:ga"```
```HOME_URL="http://aws.amazon.com/amazon-linux-ami/"```

**Comment by bdangit** [comment]
Thanks @EZtheOG, let me dig into Ohai. Maybe there's another way to check.

**Comment by EZtheOG** [comment]
https://forums.aws.amazon.com/thread.jspa?threadID=146805

It seems that it is a blend -- though, I usually download Centos6-based packages as they seem to work.

**Comment by bdangit** [comment]
@EZtheOG sorry for the long drawn out back and forth conversation on this.  I did some research and it looks like from reading [amazon ami faq](https://aws.amazon.com/amazon-linux-ami/faqs/), it appears their image is based of RHEL / Fedora 6.  

This is even a bit more evidence:

```
[ec2-user@ip-172-30-0-86 yum.repos.d]$ cat epel.repo
[epel]
name=Extra Packages for Enterprise Linux 6 - $basearch
#baseurl=http://download.fedoraproject.org/pub/epel/6/$basearch
mirrorlist=https://mirrors.fedoraproject.org/metalink?repo=epel-6&arch=$basearch
failovermethod=priority
enabled=0
gpgcheck=1
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-6
```

Checking what `ohai` provides, it would be this:

https://github.com/chef/ohai/blob/8aa8eec09148a1a2b0df540f6a174c7b50ea6feb/spec/unit/plugins/linux/platform_spec.rb#L110-L115

So with that said, there's 2 routes we can take: 

1. Update logic and assume if you are `node['platform'].eql? "amazon"` then make sure to get the release6.

or

2. You can use the new `file` install type that @chrisduong has helped to add in.  You would be in control as to what version and from where it can come from.

Let me know what you think.

**Comment by bdangit** [comment]
ref: file install type support: https://github.com/bdangit/chef-influxdb/pull/115

**Comment by bdangit** [comment]
PR #124 will help to provide the case for Amazon to get the right repo setup.

**Comment by bdangit** [comment]
influxdb 5.0.0 is merged and uploaded to supermarket.  have fun cooking!
